tests/src/Functional/CorporateFooterRenderTest.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Drupal\Tests\oe_corporate_blocks\Functional;
+
+use Drupal\Tests\BrowserTestBase;
+use Symfony\Component\DomCrawler\Crawler;
+
+/**
+ * Test footer block rendering.
+ */
+class CorporateFooterRenderTest extends BrowserTestBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  protected static $modules = [
+    'block',
+    'oe_corporate_blocks',
+    'oe_corporate_site_info',
+    'system',
+    'user',
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  protected $defaultTheme = 'stark';
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp(): void {
+    parent::setUp();
+
+    \Drupal::configFactory()
+      ->getEditable('oe_corporate_site_info.settings')
+      ->set('accessibility', 'https://example.com/accessibility')
+      ->save();
+  }
+
+  /**
+   * Tests EC footer block rendering.
+   */
+  public function testEcFooterBlockRendering(): void {
```

Comment:
These two methods are completely the same except for the block being tested, so either we use a data provider or we make a loop (preferred). Using the loop saves us install time.
You can wrap the loop in a try/catch statement like it's [done here](https://github.com/openeuropa/oe_bootstrap_theme/blob/e4fb81f7032e5f5c1648dc03739e3c950bea9f61/tests/src/PatternAssertion/CarouselPatternAssert.php#L124) so that we know in which scenario it has failed.